### ISSUE TITLE
CRAN warnings in r-devel-windows-x86_64

### ISSUE THREAD

**Issue by nacnudus** [tidyverse/readr#1528]
The [CRAN checks](https://cran.r-project.org/web/checks/check_results_readr.html) give several warnings about `src/unicode_fopen.h`.

> Version: 2.1.4
> Check: whether package can be installed
> Result: WARN 
>   Found the following significant warnings:
>     unicode_fopen.h:37:52: warning: conversion lacks type at end of format [-Wformat=]
>     unicode_fopen.h:37:14: warning: too many arguments for format [-Wformat-extra-args]
>     unicode_fopen.h:59:52: warning: conversion lacks type at end of format [-Wformat=]
>     unicode_fopen.h:59:14: warning: too many arguments for format [-Wformat-extra-args]
>   See 'd:/Rcompile/CRANpkg/local/4.4/readr.Rcheck/00install.out' for details.
>   * used C compiler: 'gcc.exe (GCC) 12.3.0'
>   * used C++ compiler: 'g++.exe (GCC) 12.3.0'
> Flavor: [r-devel-windows-x86_64](https://www.r-project.org/nosvn/R.check/r-devel-windows-x86_64/readr-00check.html)

The [same warnings](https://cran.r-project.org/web/checks/check_results_meltr.html) occur for https://github.com/r-lib/meltr, which I maintain, and which includes the same file `src/unicode_fopen.h` from when meltr was split from this repo. I don't have a fix, and would be grateful to follow anything done in this repo.

**Comment by jennybc** [comment]
Yeah I have an email from CRAN. The fix will presumably be along the same lines as what I recently did in vroom:

https://github.com/tidyverse/vroom/commit/997e9458fa5a57b98024628e96b848ff9a079dff

**Comment by jennybc** [comment]
What I see on CRAN as of 2024-01-09:

```
Version: 2.1.4
Check: whether package can be installed
Result: WARN 
  Found the following significant warnings:
    unicode_fopen.h:37:52: warning: conversion lacks type at end of format [-Wformat=]
    unicode_fopen.h:37:14: warning: too many arguments for format [-Wformat-extra-args]
    unicode_fopen.h:59:52: warning: conversion lacks type at end of format [-Wformat=]
    unicode_fopen.h:59:14: warning: too many arguments for format [-Wformat-extra-args]
  See 'd:/Rcompile/CRANpkg/local/4.4/readr.Rcheck/00install.out' for details.
  * used C compiler: 'gcc.exe (GCC) 12.3.0'
  * used C++ compiler: 'g++.exe (GCC) 12.3.0'
Flavor: [r-devel-windows-x86_64](https://www.r-project.org/nosvn/R.check/r-devel-windows-x86_64/readr-00check.html)
```

Pertinent details from the email:

> In particular look at the Windows issues.
>
> Please correct before 2024-01-18 to safely retain your package on CRAN.

**Comment by jennybc** [comment]
Fixed in 5afd348e72ad142d5865d4e5d5b19178e360b551.

Proof that it is fixed is in unmerged PR that adds a Windows r-devel GHA job to show going from red to green: #1529
